### PR TITLE
Use save not insert when storing output

### PR DESCRIPTION
**Pull request by sajal**
This is the default behavior i expected from this lib...  Took me a while to figure out why i wasn't seeing updated results.

In case this change is not acceptable, i can make it configurable.

### CODE REVIEW

**Comment by sajal** [comment]
Currently when using create_input_splits: False , MongoDisco forgets the query. The second commit fixes that.

**Comment by sajal** [comment]
re: d040cea
Before:-

`config = {
    "input_uri": "mongodb://localhost/test.in",
    "output_uri": "mongodb://localhost/test.out",
    "create_input_splits": True,
    "split_key": {'timestamp':1},
    "query": {'timestamp': {"$gt" : datetime(2012, 10, 1), "$lt" : datetime(2012, 10, 10) } }
}`

Now my collection contains loads of data. This would make splits of the whole collection and send to mapper...
After this change:-

`config = {
    "input_uri": "mongodb://localhost/test.in",
    "output_uri": "mongodb://localhost/test.out",
    "create_input_splits": True,
    "split_key": {'timestamp':1},
    "query": {'timestamp': {"$gt" : datetime(2012, 10, 1), "$lt" : datetime(2012, 10, 10) } },
    "split_max": {'timestamp': datetime(2012, 10, 10) },
    "split_min": {'timestamp': datetime(2012, 10, 1) }
}`

Now the splits are pre-limited.

**Comment by aflock** [comment]
@sajal this looks like an acceptable change, but should most likely be configurable (can choose between saving and insert)

In any case, I'd ask you to submit the pull request to the 10gen fork of this project as it is more up to date and more regularly maintained

https://github.com/mongodb/mongo-disco

cheers

**Comment by sajal** [comment]
@aflock  will do. I agree on the configurable option between save, insert and update(i just realized i need to update only some fields in one current usecase i have).

File: mongodisco/splitter.py [hunk 1]
```diff
@@ -109,6 +109,12 @@ def calculate_unsharded_splits(config, uri):
     cmd["keyPattern"]   = split_key
     cmd["force"]        = False
 
+    split_max = config.get('split_max', None)
```
Comment (aflock) [inline mongodisco/splitter.py]: Probably don't need to specify `None`, as it's the default for `.get`

Comment (sajal) [inline mongodisco/splitter.py]: aha yeah... forgot about that.

File: mongodisco/splitter.py [hunk 2]
```diff
@@ -109,6 +109,12 @@ def calculate_unsharded_splits(config, uri):
...
+    split_min = config.get('split_min', None)
+    if split_min is not None and split_max is not None:
```
Comment (aflock) [inline mongodisco/splitter.py:114]: Perhaps this is a bit more readable as
`if split_min and split_max` 
However,
Is the expected behavior that if only one of `split_max` or `split_min` is defined, the command dictionary receives neither?
If it's feasible to have either or both defined, we could change this whole block to

```
cmd["min"] = config.get('split_min')
cmd["max"] = config.get('split_max')
```

**Comment by aflock** [comment]
made a few notes- glad you are finding the project useful and thanks for submitting this request

**Comment by sajal** [comment]
https://github.com/mongodb/mongo-disco was the repository i had originally tried to use, but i found bugs in it.. 

https://github.com/mongodb/mongo-disco/blob/master/mongodisco/splitter.py#L40

it gets config into isSharded , useShards, useChunks variables, but uses is_sharded, use_shards, use_chunks in conditions.. that led me to think that its an unmaintained fork of this repo.

Are you sure I should push to https://github.com/mongodb/mongo-disco ?

File: mongodisco/splitter.py (hunk 2 above)
Comment (sajal) [inline mongodisco/splitter.py:114]: From my testing today, mongo gives error if only one of min or max is specified. It needs both to be set, or both unset. Cant set only one.

**Comment by aflock** [comment]
Hmm that's a good catch- this was a project we worked on with 10gen that was handed over to them in May, but if their fork is buggy, I'm happy to merge in your changes to this one. For now, I'd use your own fork. If you make the option to save/insert configurable and test it, I'll merge this pull request.

File: mongodisco/splitter.py (hunk 2 above)
Comment (aflock) [inline mongodisco/splitter.py:114]: got it, this LGTM then

**Comment by sajal** [comment]
@aflock : Ive made it configurable now. and have been using it for some days now.

Config variables:-
add_action : defaults to insert. Other options save and update
add_upsert: If add_action is update, then add_upsert is the boolean flag to decide if upsert should be true or false. default is false

If the action is update, it val should contain $set, $inc, $unset, etc, basically the document portion http://api.mongodb.org/python/current/api/pymongo/collection.html#pymongo.collection.Collection.update

File: mongodisco/splitter.py [hunk 3]
```diff
@@ -109,6 +109,12 @@ def calculate_unsharded_splits(config, uri):
     cmd["keyPattern"]   = split_key
     cmd["force"]        = False
 
+    split_max = config.get('split_max')
+    split_min = config.get('split_min')
+    if split_min is not None and split_max is not None:
```
Comment (aflock) [inline mongodisco/splitter.py:114]: Let's just change this to `if split_min and split_max` ? A bit more readable I think

**Comment by aflock** [comment]
Great man - apart from my one comment this looks good; I'll merge it in later today if I get a second

**Pull request merged**